### PR TITLE
Override enforcement on calls to `get_user` when the requesting user is a member of the same project as the target user

### PR DESCRIPTION
**Pull request by super-cooper**

### CODE REVIEW

**Comment by super-cooper** [comment]
[Show which user created entities in Horizon](https://trello.com/c/HKbLpzyW/2830-show-which-user-created-entities-in-horizon)

File: keystone/api/users.py [hunk 1]
```diff
@@ -187,13 +186,42 @@ def _get_user(self, user_id):
...
         GET/HEAD /v3/users/{user_id}
         """
-        ENFORCER.enforce_call(
-            action='identity:get_user',
-            build_target=_build_user_target_enforcement
-        )
+        # As a particular override for Chameleon, we allow users to get
+        # other users in their same project.
+        if self._in_same_project_as_user(user_id):
+            ks_flask.common.set_unenforced_ok()
```
Comment (Mark-Powers) [inline keystone/api/users.py:192]: Why exactly is this required? It seems to me you could just skip the ENFORCER call in the else branch.

**Review by Mark-Powers — APPROVED** [review]
Looks good!

File: keystone/api/users.py (hunk 1 above)
Comment (super-cooper) [reply keystone/api/users.py:192]: All API calls require a call to enforcement, or explicitly stating that it's unenforced. Without the this call, it throws.

**Comment by msherman64** [comment]
Nicely simple change!

I'm wondering if we should start documenting what/why/where we've forked things. If we accidentally drop this commit during a later rebase/upgrade, this seems like it would be super frustrating to debug.

**Comment by super-cooper** [comment]
As discussed this morning, this would be a good thing for all our forks. For right now, this change should be easy to identify because this fork differs minimally from base Keystone.

**Review by msherman64 — COMMENTED** [review]
+1

**Pull request merged**